pkg/logs/interface.go

```diff
@@ -0,0 +1,9 @@
+package logs
+
+import "io"
+
+type Client interface {
+	// DevModeLogs gets logs for the provided component name and namespace. A component could have multiple pods and
+	// containers running on the cluster. It returns a map where container name is the key and its logs are the value
```

Comment:
You will probably have several containers from different pods with same name

Reply:
Yep. How do we want to handle such a situation? 

We didn't discuss/imagine this scenario earlier. Do you think it would be rare enough to fix when some user opens an issue about it? 

Or, we could make pod name + container name as the key. However, odo users might not really know what a pod is, no? They are generally enterprise devs good at some framework/language, and not at k8s. It would also be ugly because odo creates a deployment which creates pod. For example, I have below pods created using the devfile I have mentioned in PR:
```sh
$ kubectl get pods
NAME                                  READY   STATUS    RESTARTS       AGE
infinitedeployment-67899ff5c5-jppvz   2/2     Running   0              31s
infinitepodone                        1/1     Running   0              31s
infinitepodtwo                        1/1     Running   4 (126m ago)   6d4h
node-app-64c7b9c8fc-kglhz             1/1     Running   0              31s
```

Again, pod is the most basic unit of deployment for k8s, so one could argue that someone dealing with k8s in any form would know what a pod is.

cc @kadel

Reply:
> We didn't discuss/imagine this scenario earlier. Do you think it would be rare enough to fix when some user opens an issue about it?

We need to handle this scenario. Sooner or later it will be a problem and we should have a solution from the beginning. I can see situations where for example the container for the inner loop and the outer loop would be named the same way. 

> Again, pod is the most basic unit of deployment for k8s, so one could argue that someone dealing with k8s in any form would know what a pod is.

We should not try to hide the fact that there are Kubernetes resources underneath.  We are already exposing users to Pod, Deployments terms in other parts like `ode delete` and it is completely fine. 



For the inner loop we should be using name of the container from the Devfile container component (`components.container.name`). Devfile spec guarantees that there can't be two components with the same name.

With the outer loop resources, it is more complicated. I wouldn't be afraid to show Deployment name / Pod name / container name. But that could get quite long. @feloy proposal could solve this problem here.